### ISSUE TITLE
[REQ] Add adjoint for out-of-place tile Cholesky factorization

### ISSUE THREAD

**Issue by ns5678** [NVIDIA/warp#1316]
### Description
Add backward pass support for out-of-place `wp.tile_cholesky()`, enabling gradient computation through Cholesky factorization via `wp.Tape()`.

### Context
Currently `wp.tile_cholesky()` is marked as ``is_differentiable=False``, which limits its use in differentiable pipelines.

**Comment by shi-eric** [comment]
Implemented in commit [`d683d4d`](https://github.com/NVIDIA/warp/commit/d683d4d654c58a275a07545dd0b60f7b31bc3f94). `wp.tile_cholesky()` (out-of-place variant) is now differentiable and can be used under `wp.Tape()`.

```python
@wp.kernel
def cholesky_kernel(A: wp.array2d(dtype=float),
                    L: wp.array2d(dtype=float)):
    a = wp.tile_load(A, shape=(N, N))
    l = wp.tile_cholesky(a)
    wp.tile_store(L, l)

A = wp.array(A_np, requires_grad=True)
L = wp.zeros((N, N), dtype=float, requires_grad=True)

with wp.Tape() as tape:
    wp.launch_tiled(cholesky_kernel, dim=[1], inputs=[A, L], block_dim=64)

tape.backward(grads={L: wp.array(adj_L_np)})
# A.grad now contains the symmetric-matrix Cholesky adjoint
```

The adjoint implements the symmetric-matrix derivative of Cholesky (Murray 2016): it solves `L^T \ Phi(L^T adj_L)` via two triangular solves and accumulates gradients into the lower triangle of `A`. Three execution paths are wired up:

- **CPU / GPU-without-MathDx:** single-thread scalar back-substitution.
- **GPU + MathDx with backward disabled:** dummy LTO symbols (no-op).
- **GPU + MathDx:** LTO-accelerated GEMM + TRSM with threaded element-wise steps using shared-memory scratch.

Note that only the **out-of-place** variant got an adjoint. The in-place `tile_cholesky_inplace` form is still non-differentiable.
